test/built-ins/Object/getOwnPropertyDescriptors/proxy-undefined-descriptor.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2016 Jordan Harband. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Object.getOwnPropertyDescriptors should perform observable operations in the correct order
+esid: sec-object.getownpropertydescriptors
+author: Jordan Harband
+features: [Proxy]
+includes: [proxyTrapsHelper.js]
+---*/
+
+var key = "a";
+var ownKeys = [key];
+var badProxyHandlers = allowProxyTraps({
+  getOwnPropertyDescriptor: function () {},
+  ownKeys: function () {
+	return ownKeys;
+  }
+});
+var proxy = new Proxy({}, badProxyHandlers);
+
+var keys = Object.keys(proxy);
```

Comment:
I guess there's another nit --- Object.keys() does not simply return the result of `O.[[OwnPropertyKeys]]()`, so this will not be the same value.

Maybe test262 could benefit from a "deepEquals()" sort of assertion to simplify this sort of thing?